nomad/structs/job.go

```diff
@@ -16,6 +24,53 @@ const (
 	JobServiceRegistrationsRPCMethod = "Job.GetServiceRegistrations"
 )
 
+// JobBatchDeregisterRequest is used to batch deregister jobs and upsert
+// evaluations.
+type JobBatchDeregisterRequest struct {
+
+	// Jobs is the set of jobs to deregister.
+	Jobs map[NamespacedID]*JobDeregisterOptions
+
+	// Evals is the set of evaluations to create.
+	//
+	// Deprecated: the job batch deregister endpoint no longer generates an
+	// eval per job.
+	Evals []*Evaluation
+
+	// SubmitTime is the time at which the job was requested to be stopped.
+	//
+	// Deprecated: The job batch deregister endpoint is only used by internal
+	// garbage collection meaning the job is removed from state, and we do not
+	// need to modify the submit time.
+	SubmitTime int64
+
+	WriteRequest
+}
+
+// JobDeregisterOptions configures how a job is deregistered.
+type JobDeregisterOptions struct {
+
+	// Purge controls whether the deregister purges the job from the system or
+	// whether the job is just marked as stopped and will be removed by the
+	// garbage collector.
+	//
+	// This request option is only ever used by the internal garbage collection
+	// process, so is always set to true.
+	Purge bool
+}
+
+// JobBatchDeregisterResponse is used to respond to a batch job deregistration.
+type JobBatchDeregisterResponse struct {
+
+	// JobEvals maps the job to its created evaluation.
+	//
+	// Deprecated: The job batch deregister endpoint is only used by internal
+	// garbage collection which no longer creates and evaluation per job GC.
+	JobEvals map[NamespacedID]string
```

Comment:
I think we can safely remove this? During upgrades, a leader with the new code will send back the new struct missing the `JobEvals` field, which the caller will interpret as an empty list.

But if we're feeling paranoid about it, we could always leave a note that it can be removed in the next LTS release.

Reply:
Updated within: https://github.com/hashicorp/nomad/pull/20510/commits/039bf6aca7f8f83e5f783aa7760f33446e7cc1cd

Tested upgrades in AWS without issue.